be/src/olap/merger.cpp

```diff
@@ -385,6 +390,51 @@ Status Merger::vertical_compact_one_group(int64_t tablet_id, ReaderType reader_t
     return Status::OK();
 }
 
+int64_t estimate_batch_size(int group_index, BaseTabletSPtr tablet, int64_t way_cnt) {
+    std::unique_lock<std::mutex> lock(tablet->sample_info_lock);
+    CompactionSampleInfo info = tablet->sample_infos[group_index];
+    if (way_cnt <= 0) {
+        LOG(INFO) << "estimate batch size for vertical compaction, tablet id: "
+                  << tablet->tablet_id() << " way cnt: " << way_cnt;
+        return 4096 - 32;
+    }
+    int64_t block_mem_limit = config::compaction_memory_bytes_limit / way_cnt;
```

Comment:
There are meta consuming memory too.

Reply:
This PR temporarily does not calculate the memory consumption of metadata